SYCL/ESIMD/api/functional/ctors/operator_assignment_copy_core.cpp

```diff
@@ -0,0 +1,61 @@
+//==------- operator_assignment_copy_core.cpp  - DPC++ ESIMD on-device test
+//          ----------------------------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for simd copy assignment operator.
+// The test creates source simd instance with reference data and invokes copy
+// assignment operator from source to destination simd instance. It is expected
+// for destination simd instance to store a bitwise same data as the reference
+// one.
+
+#include "operator_assignment.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct copy_assignment {
+  static std::string get_description() { return "copy assignment operator"; }
+
+  template <typename DataT, int NumElems>
+  static bool call_simd_ctor(const DataT *const ref_data, DataT *const out) {
+    simd<DataT, NumElems> source_simd;
+    source_simd.copy_from(ref_data);
+    simd<DataT, NumElems> simd_obj;
+    simd_obj = source_simd;
+    simd_obj.copy_to(out);
+    return simd_obj.get_test_proxy().was_move_destination();
```

Comment:
```suggestion
    return simd_obj.get_test_proxy().was_move_destination() == false;
```
No need to set `false` for `copy` and `true` for `move` on the upper level of abstraction, right? Don't forget to rename `was_moved` in `run_test` appropriately

Reply:
I've done it [here](https://github.com/intel/llvm-test-suite/pull/762/commits/afbf175e94a84d6a621ba620d2c0119abd0ec6aa#diff-4f695945bc5522c650a312a3ef5b0a9fbacce7dc658eb84d8ebad6754bff46a6R40)